uiautomationutilities/src/main/java/com/microsoft/identity/client/ui/automation/interaction/b2c/IdLabB2cSisoPolicyPromptHandler.java

```diff
@@ -0,0 +1,79 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.client.ui.automation.interaction.b2c;
+
+import androidx.annotation.NonNull;
+
+import com.microsoft.identity.client.ui.automation.interaction.AadLoginComponentHandler;
+import com.microsoft.identity.client.ui.automation.interaction.AbstractPromptHandler;
+import com.microsoft.identity.client.ui.automation.interaction.ILoginComponentHandler;
+import com.microsoft.identity.client.ui.automation.utils.UiAutomatorUtils;
+import com.microsoft.identity.internal.testutils.labutils.LabConstants;
+
+/**
+ * A Prompt handler for MSIDLAB B2C SISO Policy
+ */
+public class IdLabB2cSisoPolicyPromptHandler extends AbstractPromptHandler {
+
+    public IdLabB2cSisoPolicyPromptHandler(@NonNull final B2CPromptHandlerParameters parameters) {
+        super(getAppropriateLoginComponentHandler(parameters), parameters);
+    }
+
+    @Override
+    public void handlePrompt(@NonNull final String username, @NonNull final String password) {
+        final B2CPromptHandlerParameters b2CPromptHandlerParameters =
+                (B2CPromptHandlerParameters) parameters;
+
+        final B2CProvider b2CProvider = b2CPromptHandlerParameters.getB2cProvider();
+
+        if (b2CProvider.isExternalIdp()) {
+            assert b2CProvider.getIdpSelectionBtnResourceId() != null;
+            UiAutomatorUtils.handleButtonClick(b2CProvider.getIdpSelectionBtnResourceId());
+        }
+
+        if (!b2CProvider.isExternalIdp() && !parameters.isLoginHintProvided()) {
+            loginComponentHandler.handleEmailField(username);
+        } else if (b2CProvider.isExternalIdp() && !parameters.isSessionExpected()) {
+            loginComponentHandler.handleEmailField(username);
+        }
```

Comment:
|| ?

Reply:
It needs to be an && (at least for this policy)

Reply:
What I was saying here was:

if((!b2CProvider.isExternalIdp() && !parameters.isLoginHintProvided()) ||
(b2CProvider.isExternalIdp() && !parameters.isSessionExpected())) {
loginComponentHandler.handleEmailField(username);
}

Reply:
Addressed here: https://github.com/AzureAD/microsoft-authentication-library-common-for-android/pull/939/commits/87dcc7432872aa5b19ce4d88d395c9c28f79a591